utilities/test_suite/HIP/Tensor_hip.cpp

```diff
@@ -0,0 +1,1877 @@
+            {
+                *outputTemp = (Rpp8u)RPPPIXELCHECK(*outputf32Temp * invConversionFactor);
+                outputf32Temp++;
+                outputTemp++;
+            }
+        }
+        else if ((inputBitDepth == 5) || (inputBitDepth == 6))
+        {
+            hipMemcpy(output, d_output, outputBufferSize, hipMemcpyDeviceToHost);
+            Rpp8u *outputTemp = outputu8 + dstDescPtr->offsetInBytes;
+            Rpp8s *outputi8Temp = (Rpp8s *)output + dstDescPtr->offsetInBytes;
+            for (int i = 0; i < oBufferSize; i++)
+            {
+                *outputTemp = (Rpp8u) RPPPIXELCHECK(((Rpp32s) *outputi8Temp) + 128);
+                outputi8Temp++;
+                outputTemp++;
+            }
+        }
+
+        // Calculate exact dstROI in XYWH format for OpenCV dump
+        if (roiTypeSrc == RpptRoiType::LTRB)
+        {
+            for (int i = 0; i < dstDescPtr->n; i++)
+            {
+                int ltX = roiTensorPtrSrc[i].ltrbROI.lt.x;
+                int ltY = roiTensorPtrSrc[i].ltrbROI.lt.y;
+                int rbX = roiTensorPtrSrc[i].ltrbROI.rb.x;
+                int rbY = roiTensorPtrSrc[i].ltrbROI.rb.y;
+
+                roiTensorPtrSrc[i].xywhROI.xy.x = ltX;
+                roiTensorPtrSrc[i].xywhROI.xy.y = ltY;
+                roiTensorPtrSrc[i].xywhROI.roiWidth = rbX - ltX + 1;
+                roiTensorPtrSrc[i].xywhROI.roiHeight = rbY - ltY + 1;
+            }
+        }
+
+        if(inputBitDepth == 0 && (srcDescPtr->layout == dstDescPtr->layout))
+            compare_output<Rpp8u>(outputu8, func, testCaseName, dstDescPtr, "HIP");
+
+        RpptROI roiDefault;
+        RpptROIPtr roiPtrDefault;
+        roiPtrDefault = &roiDefault;
+        roiPtrDefault->xywhROI.xy.x = 0;
+        roiPtrDefault->xywhROI.xy.y = 0;
+        roiPtrDefault->xywhROI.roiWidth = dstDescPtr->w;
+        roiPtrDefault->xywhROI.roiHeight = dstDescPtr->h;
+
+        for (int i = 0; i < dstDescPtr->n; i++)
+        {
+            roiTensorPtrSrc[i].xywhROI.roiWidth = RPPMIN2(roiPtrDefault->xywhROI.roiWidth - roiTensorPtrSrc[i].xywhROI.xy.x, roiTensorPtrSrc[i].xywhROI.roiWidth);
+            roiTensorPtrSrc[i].xywhROI.roiHeight = RPPMIN2(roiPtrDefault->xywhROI.roiHeight - roiTensorPtrSrc[i].xywhROI.xy.y, roiTensorPtrSrc[i].xywhROI.roiHeight);
+            roiTensorPtrSrc[i].xywhROI.xy.x = RPPMAX2(roiPtrDefault->xywhROI.xy.x, roiTensorPtrSrc[i].xywhROI.xy.x);
+            roiTensorPtrSrc[i].xywhROI.xy.y = RPPMAX2(roiPtrDefault->xywhROI.xy.y, roiTensorPtrSrc[i].xywhROI.xy.y);
+        }
+
+        // Convert any PLN3 outputs to the corresponding PKD3 version for OpenCV dump
+        if (layoutType == 0 || layoutType == 1)
+        {
+            if ((dstDescPtr->c == 3) && (dstDescPtr->layout == RpptLayout::NCHW))
+                convert_pln3_to_pkd3(outputu8, dstDescPtr);
+        }
+        rppDestroyGPU(handle);
+
+        // OpenCV dump (if testType is unit test)
```

Comment:
Pls compare the OpenCV dump part of the code line by line in host and hip. If they are the same, lets modularize it

Reply:
Done